src/hooks.server.js

```diff
@@ -37,15 +37,33 @@ async function getUserDetails({event, resolve}){
 
   // console.log({reqUrl: event.url.pathname, user: event.locals?.user, gettingDetails: event.locals.session && ! event.locals?.user && ! event.route.id?.includes('api/home') && ! event.request.url.includes('assets')});
   if (event.locals.session && ! event.locals?.user && ! event.route.id?.includes('api/home') && ! event.request.url.includes('assets')) {
-		const getUserDetails = await api({
-			method: 'get',
-			resource: 'user',
-			event,
-		});
-
-    if (getUserDetails?.status == 200) {
-      //TODO: Set a localStorage with key user and expiration time for 5mins. If that key is present, no need to getUserDetails. @see https://www.sohamkamani.com/javascript/localstorage-with-ttl-expiry/
-      event.locals.user = await getUserDetails?.json() //use this to determine auth on frontend. Before accessing auth routes if this is null redirect to login page
+    // set time, this can also be easily changed to days, hours or month
+    const date = new Date().setMinutes(new Date().getMinutes() + 5); //change this to increase tht expiring date
+    
+    const user = sessionStorage.getItem('user');
+
+    if(!user){
+      const getUserDetails = await api({
+        method: 'get',
+        resource: 'user',
+        event,
+      });
+  
+      if (getUserDetails?.status == 200) {
+        //TODO: Set a localStorage with key user and expiration time for 5mins. If that key is present, no need to getUserDetails. @see https://www.sohamkamani.com/javascript/localstorage-with-ttl-expiry/
+        event.locals.user = await getUserDetails?.json(); //use this to determine auth on frontend. Before accessing auth routes if this is null redirect to login page
+
+        sessionStorage.setItem('user', JSON.stringify({
+          value: event.locals.user,
+          expire_date: date,
+        }))
+      }
+    }else {
+      if(new Date().getMinutes() > JSON.parse(user).expire_date){
```

Comment:
This section will always fail. 

`new Date().setMinutes(new Date().getMinutes() + 5)` returns a timestamp like `1730359861097`, meanwhile `new Date().getMinutes()` returns the minute portion of the current time, like `8:30` will return `30`.

So this is like saying `if(30 > 1730359861097)`.